src/PrestaShopBundle/Resources/config/services/bundle/controller.yml

```diff
@@ -142,5 +142,14 @@ services:
       multistoreFeature: "@prestashop.adapter.multistore_feature"
       multistoreContext: "@prestashop.adapter.shop.context"
       entityManager: "@doctrine.orm.entity_manager"
+      productRepository: '@PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository'
+    tags:
+      - { name: !php/const PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController::PRESTASHOP_CORE_CONTROLLERS_TAG }
+
+  PrestaShopBundle\Controller\Admin\Sell\Catalog\Product\ProductController:
+    parent: PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController
+    public: true
+    properties:
+      productRepository: '@PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository'
```

Comment:
Can you please use the injection by constructor instead, this DI by properties is not ideal as it forces having public properties and doing the same thing via constructor is not that difficult (don't forget to call `parent::_construct()`)